builtin/providers/aws/resource_aws_elasticsearch_domain.go

```diff
@@ -37,13 +37,9 @@ func resourceAwsElasticSearchDomain() *schema.Resource {
 				ForceNew: true,
 				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
 					value := v.(string)
-					if !regexp.MustCompile(`^[0-9A-Za-z]+`).MatchString(value) {
+					if !regexp.MustCompile(`^[a-z][0-9a-z\-]{2,27}$`).MatchString(value) {
```

Comment:
http://docs.aws.amazon.com/sdk-for-go/api/service/elasticsearchservice.html#type-CreateElasticsearchDomainInput

> Domain names must start with a letter or number and can contain the following characters: a-z (lowercase), 0-9, and - (hyphen).

The text from the actual console is:

> The name must start with a lowercase alphabet and be at least 3 and no more than 28 characters long. Valid characters are a-z (lowercase letters), 0-9, and - (hyphen).

And when I try to start with a number it doesn't work.

Feel free to cherry pick around this, or I can rip it out, I just hit this bug as well.